common/src/test/groovy/com/mx/path/core/common/connect/RetryConfigurationTest.groovy

```diff
@@ -0,0 +1,165 @@
+    RetryConfiguration.builder().stopStrategy(RetryConfiguration.StopStrategy.COUNT).count(1).pauseStrategy(RetryConfiguration.PauseStrategy.FIXED).build()        | "Missing required fields for pauseStrategy FIXED: pause at configuration."
+    RetryConfiguration.builder().stopStrategy(RetryConfiguration.StopStrategy.COUNT).count(1).pauseStrategy(RetryConfiguration.PauseStrategy.INCREMENTING).build() | "Missing required fields for pauseStrategy INCREMENTING: initialPause, increment at configuration."
+    RetryConfiguration.builder().stopStrategy(RetryConfiguration.StopStrategy.COUNT).count(1).pauseStrategy(RetryConfiguration.PauseStrategy.FIBONACCI).build()    | "Missing required fields for pauseStrategy FIBONACCI: multiplier, maxPause at configuration."
+  }
+
+  def "valid configurations"() {
+    given:
+    def configurationState = mock(ConfigurationState)
+    when(configurationState.currentState()).thenReturn("configuration.")
+
+    when:
+    node.validate(configurationState)
+
+    then:
+    noExceptionThrown()
+
+    where:
+    node                                                                                                                                                                                                                             | _
+    RetryConfiguration.builder().stopStrategy(RetryConfiguration.StopStrategy.COUNT).count(1).build()                                                                                                                                  | null
+    RetryConfiguration.builder().stopStrategy(RetryConfiguration.StopStrategy.DURATION).duration(Duration.ofSeconds(1)).build()                                                                                                        | null
+    RetryConfiguration.builder().stopStrategy(RetryConfiguration.StopStrategy.COUNT).count(1).pauseStrategy(RetryConfiguration.PauseStrategy.FIXED).pause(Duration.ofMillis(10)).build()                                                | null
+    RetryConfiguration.builder().stopStrategy(RetryConfiguration.StopStrategy.COUNT).count(1).pauseStrategy(RetryConfiguration.PauseStrategy.INCREMENTING).initialPause(Duration.ofMillis(10)).increment(Duration.ofMillis(10)).build() | null
+    RetryConfiguration.builder().stopStrategy(RetryConfiguration.StopStrategy.COUNT).count(1).pauseStrategy(RetryConfiguration.PauseStrategy.FIBONACCI).maxPause(Duration.ofMillis(100)).multiplier(Duration.ofSeconds(1)).build()                       | null
+  }
+
+  def "call when all attempts fail"() {
+    given:
+    def subject = RetryConfiguration.builder()
+        .stopStrategy(RetryConfiguration.StopStrategy.COUNT)
+        .count(3)
+        .retryOn(Collections.singletonList(ResponseMatcher.builder().predicate({ t -> true }).build()))
+        .build()
+
+    when:
+    subject.call({
+      ->
+      null
+    } as Callable)
+
+    then:
+    def error = thrown(RetryException)
+    error.getNumberOfFailedAttempts() == 3
```

Comment:
can access properties directly without getters in groovy

Reply:
fixed.